dotnetapp-selfcontained/README.md

```diff
@@ -1,72 +1,62 @@
-dotnetapp-selfcontained Sample
-==============================
+# .NET Core self-contained application Docker Production Sample
 
-The dotnetapp-selfcontained sample demonstrates how you can build and run the dotnetapp sample as a [self-contained .NET Core application](https://docs.microsoft.com/en-us/dotnet/articles/core/deploying/) that relies only on an operating system image (plus dependencies). It's a good option for creating a Docker image for production.
+This .NET Core Docker sample demonstrates how to produce [self-contained applications](https://docs.microsoft.com/dotnet/core/deploying/) with Docker. It works with both Linux and Windows containers.
 
-The instructions assume that you already have [.NET Core 1.0](https://www.microsoft.com/net/download/core#/sdk) and [Git](https://git-scm.com/downloads) and [Docker](https://www.docker.com/products/docker) clients installed. They also assume you already know how to target Linux or Windows containers. Do try both image types. You need the latest Windows 10 or Windows Server 2016 to use [Windows containers](http://aka.ms/windowscontainers).
+The [sample Dockerfile](Dockerfile) creates an .NET Core application container based off of the [.NET Core Runtime Dependencies Docker base image](https://hub.docker.com/r/microsoft/dotnet/). This is the type of container you would want to use in production if you want the smallest possible container and do not see a [benefit from sharing base images between containers](https://docs.docker.com/engine/userguide/storagedriver/imagesandcontainers/). It uses the [Docker multi-stage build feature](https://github.com/dotnet/announcements/issues/18) to build the sample with the larger [.NET Core SDK Docker base image](https://hub.docker.com/r/microsoft/dotnet/) and then copies the final build result into a Docker image based on the smaller [.NET Core Docker Runtime Depedendencies base image](https://hub.docker.com/r/microsoft/dotnet/) in the case of Linux, and the [Windows Server 2016 Nano Server base OS image](https://hub.docker.com/r/microsoft/nanoserver/) in the case of Windows.
```

Comment:
`if you want the smallest possible container and do not see a benefit from sharing base images between containers` seems kind of harsh 😄 If used, you still share the OS and runtime-deps layers.